sdcm/tester.py

```diff
@@ -520,6 +529,143 @@ def _setup_ldap_roles(self, db_cluster: BaseScyllaCluster):
         node.run_cqlsh(f'ALTER ROLE \'{LDAP_USERS[0]}\' with SUPERUSER=true and password=\'{LDAP_PASSWORD}\'')
         self.params['are_ldap_users_on_scylla'] = True
 
+    @property
+    def ldap_ip(self) -> str:
+        ldap_address = list(self.test_config.LDAP_ADDRESS).copy()
+        return ldap_address[0]
+
+    @property
+    def ldap_port(self) -> str:
+        ldap_address = list(self.test_config.LDAP_ADDRESS).copy()
+        return ldap_address[1]
+
+    def _add_ldap_entry(self, ldap_entry: list):
+        self.log.debug("Adding an Ldap entry of: %s", ldap_entry)
+        username = f'cn=admin,{LDAP_BASE_OBJECT}'
+        self.localhost.add_ldap_entry(ip=self.ldap_ip, ldap_port=self.ldap_port,
```

Comment:
need to confirm if it works with `MS AD`

Reply:
There might be a general existing issue, using MS AD in SCT.
The test failed with:
```
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > Traceback (most recent call last):
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >   File "/home/ubuntu/scylla-cluster-tests/sdcm/nemesis.py", line 3911, in wrapper
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >     result = method(*args[1:], **kwargs)
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >   File "/home/ubuntu/scylla-cluster-tests/sdcm/nemesis.py", line 1129, in disrupt_ldap_role_add_remove_permission
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >     self.tester.create_role_in_ldap(ldap_role_name=superuser_role, unique_members=[new_test_user, LDAP_USERS[1]])
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >   File "/home/ubuntu/scylla-cluster-tests/sdcm/tester.py", line 558, in create_role_in_ldap
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >     self._add_ldap_entry(ldap_entry=role_entry)
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >   File "/home/ubuntu/scylla-cluster-tests/sdcm/tester.py", line 545, in _add_ldap_entry
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >     self.localhost.add_ldap_entry(ip=self.ldap_ip, ldap_port=self.ldap_port,
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >   File "/home/ubuntu/scylla-cluster-tests/sdcm/utils/ldap.py", line 82, in add_ldap_entry
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >     LdapContainerMixin.create_ldap_connection(ip, ldap_port, user, password)
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >   File "/home/ubuntu/scylla-cluster-tests/sdcm/utils/decorators.py", line 67, in inner
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >     return func(*args, **kwargs)
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >   File "/home/ubuntu/scylla-cluster-tests/sdcm/utils/ldap.py", line 69, in create_ldap_connection
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >     LdapContainerMixin.ldap_server = Server(host=f'ldap://{ip}:{ldap_port}', get_info=ALL)
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >   File "/usr/local/lib/python3.10/site-packages/ldap3/core/server.py", line 118, in __init__
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  >     raise LDAPInvalidPortError('port must be an integer')
15:39:42  < t:2023-01-08 13:39:42,061 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > ldap3.core.exceptions.LDAPInvalidPortError: port must be an integer
```
I think we can leave this mode out of this PR content and address it in a future PR.
Currently skipping this mode by:
```
        if not self.cluster.params.get('ldap_server_type') == LdapServerType.MS_AD:
            raise UnsupportedNemesis('Cluster is not configured to run with open LDAP authorization, hence skipping')
```